src/java.base/windows/native/libnio/ch/wepoll.c

```diff
@@ -1553,30 +1553,18 @@ bool queue_is_enqueued(const queue_node_t* node) {
 #define REFLOCK__DESTROY_MASK ((long) 0xf0000000UL)
 #define REFLOCK__POISON       ((long) 0x300dead0UL)
 
-static HANDLE reflock__keyed_event = NULL;
-
-int reflock_global_init(void) {
-  NTSTATUS status = NtCreateKeyedEvent(
-      &reflock__keyed_event, KEYEDEVENT_ALL_ACCESS, NULL, 0);
-  if (status != STATUS_SUCCESS)
-    return_set_error(-1, RtlNtStatusToDosError(status));
-  return 0;
-}
 
 void reflock_init(reflock_t* reflock) {
   reflock->state = 0;
 }
 
 static void reflock__signal_event(void* address) {
-  NTSTATUS status =
-      NtReleaseKeyedEvent(reflock__keyed_event, address, FALSE, NULL);
-  if (status != STATUS_SUCCESS)
-    abort();
+  WakeByAddressSingle(address);
 }
 
 static void reflock__await_event(void* address) {
   NTSTATUS status =
-      NtWaitForKeyedEvent(reflock__keyed_event, address, FALSE, NULL);
```

Comment:
It looks like this wait was "not alertable" (passing false here.)  https://learn.microsoft.com/en-us/windows/win32/api/synchapi/nf-synchapi-waitonaddress#remarks says "the wait may return for other reasons" ... does the code need to change to deal with that behavior?